experimental/dds/tree2/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -157,278 +156,126 @@ function composeMarkLists<TNodeChange>(
+			return baseMark;
+		}
+		return withNodeChange(baseMark, nodeChange);
+	} else if (areInputCellsEmpty(baseMark)) {
+		const moveInId = getMarkMoveId(baseMark);
+		const moveOutId = getMarkMoveId(newMark);
+
+		if (moveInId !== undefined && moveOutId !== undefined) {
+			assert(isMoveMark(baseMark) && isMoveMark(newMark), "Only move marks have move IDs");
+			const srcEffect = getOrAddEffect(
+				moveEffects,
+				CrossFieldTarget.Source,
+				baseMark.revision,
+				baseMark.id,
+				true,
+			);
 
-	const newMarkRevision = isModify(newMark) ? newRev : newMark.revision ?? newRev;
-	const newIntention = getIntention(newMarkRevision, revisionMetadata);
-	const baseMarkRevision = isModify(baseMark) ? undefined : baseMark.revision;
-	const baseIntention = getIntention(baseMarkRevision, revisionMetadata);
-
-	switch (baseType) {
-		case "Insert":
-		case "Revive":
-			switch (newType) {
-				case "Modify": {
-					return mergeInNewChildChanges(baseMark, newMark.changes, newRev, composeChild);
-				}
-				case "Delete": {
-					// The insertion made by the base change is subsequently deleted.
-					// TODO: preserve the insertions as conflicted.
-					return 0;
-				}
-				case "MoveOut":
-				case "ReturnFrom":
-					// The insert has been moved by `newMark`.
-					// We can represent net effect of the two marks as an insert at the move destination.
-					getOrAddEffect(
-						moveEffects,
-						CrossFieldTarget.Destination,
-						newMark.revision ?? newRev,
-						newMark.id,
-						true,
-					).mark = mergeInNewChildChanges(
-						baseMark,
-						newMark.changes,
-						newMark.revision ?? newRev,
-						composeChild,
-					);
-					return 0;
-				case "Revive": {
-					assert(
-						!isConflictedReattach(baseMark) && isConflicted(newMark),
-						0x4dd /* Invalid mark overlap */,
-					);
-					return baseMark;
-				}
-				default:
-					fail("Not implemented newType");
-			}
-		case "Modify": {
-			switch (newType) {
-				case "Modify": {
-					return mergeInNewChildChanges(baseMark, newMark.changes, newRev, composeChild);
-				}
-				case "Delete": {
-					// For now the deletion obliterates all other modifications.
-					// In the long run we want to preserve them.
-					return composeMark(newMark, newRev, composeChild);
-				}
-				case "MoveOut":
-				case "ReturnFrom": {
-					return composeWithBaseChildChanges(
-						newMark,
-						newRev,
-						baseMark.changes,
-						composeChild,
-					);
-				}
-				default:
-					fail("Not implemented newType");
+			const dstEffect = getOrAddEffect(
+				moveEffects,
+				CrossFieldTarget.Destination,
+				newMark.revision ?? newRev,
+				newMark.id,
+				true,
+			);
+
+			const baseIntention = getIntention(baseMark.revision, revisionMetadata);
+			const newIntention = getIntention(newMark.revision ?? newRev, revisionMetadata);
+			if (areInverseMoves(baseMark, baseIntention, newMark, newIntention)) {
+				srcEffect.shouldRemove = true;
+				dstEffect.shouldRemove = true;
```

Comment:
In this case, what happens to the `nodeChange`?

Reply:
Good point. I suppose this assumes that inverse moves also have inverse node changes. Before this PR we were discarding any node changes whenever we composed inverse marks. I'll file a bug and fix this later.

Reply:
Plz add comment referencing the bug.